src/spatialdata/_core/data_extent.py

```diff
@@ -1,36 +1,325 @@
+    ----------
+    e
+        The SpatialData object.
+
+    Returns
+    -------
+    The bounding box description.
+    """
+    new_min_coordinates_dict: dict[str, list[float]] = defaultdict(list)
+    new_max_coordinates_dict: dict[str, list[float]] = defaultdict(list)
+    mask = [has_images, has_labels, has_points, has_shapes]
+    include_spatial_elements = ["images", "labels", "points", "shapes"]
+    include_spatial_elements = [i for (i, v) in zip(include_spatial_elements, mask) if v]
+
+    if elements is None:  # to shut up ruff
+        elements = []
+    if not isinstance(elements, list):
+        raise ValueError(f"Invalid type of `elements`: {type(elements)}, expected `list`.")
+
+    for element in e._gen_elements():
+        element_type, element_name, element_obj = element
+        plot_element = (len(elements) == 0) or (element_name in elements)
+        plot_element = plot_element and (element_type in include_spatial_elements)
+        if plot_element:
+            transformations = get_transformation(element_obj, get_all=True)
+            assert isinstance(transformations, dict)
+            coordinate_systems = list(transformations.keys())
+            if coordinate_system in coordinate_systems:
+                extent = get_extent(element_obj, coordinate_system=coordinate_system)
+                axes = list(extent.keys())
+                for ax in axes:
+                    new_min_coordinates_dict[ax] += [extent[ax][0]]
+                    new_max_coordinates_dict[ax] += [extent[ax][1]]
+                if len(axes) == 0:
+                    raise ValueError(
+                        f"The SpatialData object does not contain any element in the "
+                        f" coordinate system {coordinate_system!r}, "
+                        f"please pass a different coordinate system wiht the argument 'coordinate_system'."
+                    )
+    new_min_coordinates = np.array([min(new_min_coordinates_dict[ax]) for ax in axes])
+    new_max_coordinates = np.array([max(new_max_coordinates_dict[ax]) for ax in axes])
+    extent = {}
+    for idx, ax in enumerate(axes):
+        extent[ax] = (new_min_coordinates[idx], new_max_coordinates[idx])
+    return extent
+
+
+@get_extent.register
+def _(e: GeoDataFrame, coordinate_system: str = "global") -> BoundingBoxDescription:
     """
-    circle_element = shapes
-    if not isinstance(circle_element.geometry.iloc[0], Point):
-        raise NotImplementedError("Only circles (shapely Point) are currently supported (not Polygon or MultiPolygon).")
-    circle_dims = get_axes_names(circle_element)
+    Compute the extent (bounding box) of a set of shapes.
+
+    Returns
+    -------
+    The bounding box description.
+    """
+    _check_element_has_coordinate_system(element=e, coordinate_system=coordinate_system)
+    # remove potentially empty geometries
+    e_temp = e[e["geometry"].apply(lambda geom: not geom.is_empty)]
+
+    # separate points from (multi-)polygons
```

Comment:
This is not needed because we assume that the GeoDataFrame doesn't mix Point with Polygon/MultiPolygon types. I removed it to simplify the code.

Reply:
In the PR I kept the removal of empty points, but I have removed the split of shapes and points. It can be easily restored inside the function `_get_extent_of_shapes()` if you want. We don't expect to need it now, but it would be still ok to have, just to make the code more future proof.